### ISSUE TITLE
str function should return a string primitive, not 'new String()'

### ISSUE THREAD

**Issue by GULPF** [QQuick/Transcrypt#147]
Currently, the str function looks like this:
```javascript
function str (stringable) {
  try {
    return stringable.__str__ ();
  }
  catch (exception) {
    return new String (stringable);
    }
}
```

But it would be preferable to return `String(stringable)` (without 'new'). The class-like wrappers in JS (String, Number, Boolean) behaves weirdly in a lot of cases and should be avoided. F.ex, the current behavior results in situations like this one:

```python
x = ['1', '2', '3']
print('1' in x)    # true
print(str(1) in x) # false
```

**Comment by JdeH** [comment]
So your proposal is to make it:

```
function str (stringable) {
  try {
    return stringable.__str__ ();
  }
  catch (exception) {
    return String (stringable);
    }
}
```?

Or do you mean something else, since youj recommend class-like wrappers like String to be avoided completely. If so, in your view, what would be a good alternative?

**Comment by GULPF** [comment]
Yes, that's what I mean. `String(stringable)` doesn't actually return a wrapped string, it returns a primitive. The same is true for the other wrappers, e.g `new Boolean(1)` will return a wrapped boolean, while `Boolean(1)` will return a primitive.

Of course, it's also possible to use operators for casting to primitives, like `+ ""` for strings or `!!` for booleans, both ways are fine as long as the wrappers are avoided.

**Comment by JdeH** [comment]
OK, I see. I thought I needed a wrapped string to be able to used the extra Python methods and attributes that were added to the String prototype, but from

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String

it becomes apparent  JavaScript will automatically wrap a primitive string whenever a call of a method defined on it's prototype is called. And I guess working with primitives is generally faster.

I've adapted it accordingly and the autotest runs fine. So it'll be in the next commit.
